Common/src/main/java/com/blamejared/crafttweaker/impl/registry/zencode/BracketResolverRegistry.java

```diff
@@ -0,0 +1,102 @@
+package com.blamejared.crafttweaker.impl.registry.zencode;
+
+import com.blamejared.crafttweaker.api.command.type.IBracketDumperInfo;
+import com.blamejared.crafttweaker.api.plugin.IBracketParserRegistrationHandler;
+import com.blamejared.crafttweaker.api.zencode.IScriptLoader;
+import com.mojang.datafixers.util.Pair;
+import org.openzen.zenscript.parser.BracketExpressionParser;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Contains all info on Bracket resolvers, validators and dumpers
+ */
+public final class BracketResolverRegistry {
+    
+    private record BracketData(Map<String, BracketHandle> brackets) {
+        
+        BracketData() {
+            
+            this(new HashMap<>());
+        }
+        
+    }
```

Comment:
I am not fully sure if this is actually specified anywhere. The [javadoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Record.html) for records state that:
> A *record class* is a shallowly immutable, transparent carrier for a fixed set of values, called the *record components*

While I agree it seems like a reasonable interpretation to give them full on value type semantics due to mentioning a fixed set of values (and certainly can be enforced decision wise by project maintainers), it seems to me (and from what I can find online) that most people have taken it to mean that the references to the values are fixed. This also is a somewhat more logical usage, as having them be 100% value type based would basically remove 90 or so percent of use cases of them for holding various misc objects when the object may not be immutable itself.

I also did a small test just to see whether or not things would break if records get modified this way, and at least currently:
```java
record ListTest(List<String> test) {
}
List<String> test = new ArrayList<>();
test.add("1");
ListTest listTest = new ListTest(test);
System.out.println(listTest.hashCode());
test.add("2");
System.out.println(listTest.hashCode());
```
produced the output of:
```
80
2530
```
Which means that it isn't doing any sort of caching by default that may cause records to then improperly return matching values for hashCode/equals when they don't actually match.

Reply:
Yes I know it's _possible_ from the spec side.
But from the JEP-384, the motivation is about immutable data ^^.
https://openjdk.java.net/jeps/384
```
Summary
Enhance the Java programming language with [records](https://cr.openjdk.java.net/~briangoetz/amber/datum.html),
 which are classes that act as transparent carriers for immutable data. 
Records can be thought of as nominal tuples. 
This is a [preview language feature](https://openjdk.java.net/jeps/12) in JDK 15.
```


Generally, *in my opinion,* it is a bad practive to have any object with mutable hashCode values, as it breaks some assumptions in the JDK.

E.g. 
```java
public static void main(String[] args) {
    record ListTest(List<String> list){}
    
    var test = new ListTest(new ArrayList<>());
    var map = new HashMap<ListTest, Boolean>();
    
    map.put(test, true);
    System.out.println(map.containsKey(test)); // True
    test.list.add("X");
    
    System.out.println(map.containsKey(test)); // False
}```

Reply:
After discussion on Discord, this is what we came up with: 

> So how about we decide on this then:
> \- If a record class is used only in a class, its members can be mutable
> \- If a record class is visible for other classes, its members must not be mutable
>
> As for that one record where we have both a mutable and immutable version of it, can you change it to two records, one that is used for building and one for handing it out?